codalab/worker/bundle_manager.py

```diff
@@ -0,0 +1,401 @@
+import datetime
+import os
+import re
+import subprocess
+import sys
+import time
+import traceback
+
+from codalab.common import State
+from codalab.lib import bundle_util, formatting, path_util
+from codalab.objects.permission import check_bundle_have_run_permission
+
+
+class BundleManager(object):
+    def __init__(self, codalab_manager):
+        self._model = codalab_manager.model()
+        self._worker_model = codalab_manager.worker_model()
+        self._bundle_store = codalab_manager.bundle_store()
+
+        # Parse the config.
+        config = codalab_manager.config.get('workers')
+        if not config:
+            print >> sys.stderr, 'Config is missing a workers section'
+            exit(1)
+
+        if 'default_docker_image' not in config:
+            print >> sys.stderr, 'workers config missing default_docker_image'
+            exit(1)
+        self._default_docker_image = config['default_docker_image']
+
+        if 'torque' in config:
+            assert(self._worker_model.shared_file_system)
+            self._use_torque = True
+            self._torque_host = config['torque']['host']
```

Comment:
Maybe make this more explicit - 'ssh_host' - this is simply where we ssh into to run commands.